platforms/core-configuration/file-collections/src/main/java/org/gradle/api/internal/file/AbstractFileResolver.java

```diff
@@ -71,42 +80,30 @@ public String resolveForDisplay(Object path) {
 
     @Override
     public File resolve(Object path, PathValidation validation) {
```

Comment:
Ideally, the input should also be typed. However, I think we need a decision about the future of Repository.url and similar properties before that.